src/app/app.component.ts

```diff
@@ -1,20 +1,34 @@
 ///<reference path="govuk.d.ts">
 import { Component, OnInit } from '@angular/core';
-import { select, Store } from '@ngrx/store';
+import { Router, NavigationEnd, Event, ActivatedRoute } from '@angular/router';
+import { Store, select } from '@ngrx/store';
 import { LoadingService } from '@services/loading/loading.service';
 import { UserService } from '@services/user-service/user-service';
 import { selectRouteData } from '@store/router/selectors/router.selectors';
 import { initAll } from 'govuk-frontend/govuk/all';
-import { map, take } from 'rxjs';
+import { take, map } from 'rxjs';
 import { State } from './store';
+import { GoogleAnalyticsService } from '@services/google-analytics/google-analytics.service';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['app.component.scss']
 })
 export class AppComponent implements OnInit {
-  constructor(public userService: UserService, private loadingService: LoadingService, private store: Store<State>) {}
+  constructor(
+    public userService: UserService,
+    private loadingService: LoadingService,
+    private router: Router,
+    private store: Store<State>,
+    private googleAnalyticsService: GoogleAnalyticsService
+  ) {
+    this.router.events.subscribe((event: Event) => {
```

Comment:
Probably should `takeUntil(destoy$)` here...

Reply:
This is kind of redundant as the app component is responsible for rendering the whole system and will never get destroyed. It will only be destroyed when they close the tab that runs the app, at which point the browser will release all memory for it anyway.

Reply:
I did think that, just think it's probably better practice anyways.